src/Generator/Generators/CLI/CLITypeReferences.cs

```diff
@@ -191,7 +191,8 @@ public override bool VisitClassDecl(Class @class)
             if (@class.IsIncomplete && @class.CompleteDeclaration != null)
                 @class = (Class) @class.CompleteDeclaration;
 
-            var keywords = @class.IsValueType? "value struct" : "ref class";
+            var keywords = (@class.IsValueType && !(@class.HasBaseClass && @class.BaseClass.IsRefType))
```

Comment:
@triao I have one last change for this pull. I will push that tomorrow. If
you see it before then, please don't merge it yet.